docs/src/modules/components/Demo.js

```diff
@@ -250,16 +269,25 @@ export default function Demo(props) {
         </NoSsr>
       )}
       <Collapse in={openDemoSource} unmountOnExit>
-        <Code
-          id={demoSourceId}
-          code={showPreview && !codeOpen ? demo.jsxPreview : demoData.raw}
-          language={demoData.sourceLanguage}
-          copyButtonProps={{
-            'data-ga-event-category': codeOpen ? 'demo-expand' : 'demo',
-            'data-ga-event-label': demoOptions.demo,
-            'data-ga-event-action': 'copy-click',
-          }}
-        />
+        <div>
+          <CodeEditor
+            key={demoKey}
+            id={demoSourceId}
+            value={code}
+            onChange={setCode}
+            language={demoData.sourceLanguage}
+          />
+          {debouncedError && (
```

Comment:
Nitpick: but the debounce should only happen when the error appears. As soon as the error is fixed, the error can disappear without debounce.

```suggestion
          {debouncedError && error && (
```

Reply:
The case is that every time if you updated the code, the `error` will be reset to `null`, so by doing so will cause glitch, another solution is `const debouncedSetError = (error) => debounceFn(setError, error? 300: 10)`, but on a second thought, I think the current way is fine, as in continues editing, it could change from error -> null -> error

Reply:
@Janpot you suggestion works indeed, thanks